go/registry/api/api.go

```diff
@@ -40,7 +40,7 @@ var (
 
 	// DeregisterEntitySignatureContext is the context used for entity
 	// deregistration.
-	DeregisterEntitySignatureContext = signature.NewContext("oasis-core/registry: deregister entity")
+	DeregisterEntitySignatureContext = signature.NewContext("oasis-core/registry: deregister entity", signature.WithChainSeparation())
```

Comment:
This breaks the registry cli.

Reply:
I should probably wait for #2374 then.

Do we have tests for the registry CLI? We should have something similar to how we have E2E tests for the staking CLI, I'll open an issue.

Reply:
I assume entity deregistrations don't interact with the debonding logic at all (we don't need to persist signed deregistrations across a dump/restore boundary)...

Reply:
Yeah we don't need to persist deregistrations which is why we can easily add a chain separation context.

Ideally we could do the same for node/entity registrations, but then dump/restore wouldn't work without everyone re-signing.

Reply:
I've changed this back for now.